kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/GCSToBQLoadRunnable.java

```diff
@@ -0,0 +1,232 @@
+
+    Page<Blob> list = bucket.list();
+    for (Blob blob : list.iterateAll()) {
+      TableId table = getTableFromBlob(blob);
+
+      if (table == null || claimedBlobs.contains(blob)) {
+        // don't do anything if:
+        // 1. we don't know what table this should be uploaded to or
+        // 2. this blob is already claimed by a currently-running job.
+        continue;
+      }
+
+      if (!tableToURIs.containsKey(table)) {
+        // initialize maps, if we haven't seen this table before.
+        tableToURIs.put(table, new ArrayList<>());
+        tableToCurrentLoadSize.put(table, 0L);
+      }
+
+      long newSize = tableToCurrentLoadSize.get(table) + blob.getSize();
+      // if this file does not cause us to exceed our per-request quota limits...
+      if (newSize <= MAX_LOAD_SIZE_B && tableToURIs.get(table).size() < FILE_LOAD_LIMIT) {
+        // ...add the file (and update the load size)
+        tableToURIs.get(table).add(blob);
+        tableToCurrentLoadSize.put(table, newSize);
+      }
+    }
+    return tableToURIs;
+  }
+
+  /**
+   * Given a blob, return the {@link TableId} this blob should be inserted into.
+   * @param blob the blob
+   * @return the TableId this data should be loaded into, or null if we could not tell what
+   *         table it should be loaded into.
+   */
+  private TableId getTableFromBlob(Blob blob) {
+    if (blob.getMetadata() == null
+        || blob.getMetadata().get(GCSToBQWriter.GCS_METADATA_TABLE_KEY) == null) {
+      logger.error("Found blob {}\\{} with no metadata.", blob.getBucket(), blob.getName());
+      return null;
+    }
+
+    String serializedTableId = blob.getMetadata().get(GCSToBQWriter.GCS_METADATA_TABLE_KEY);
+    Matcher matcher = METADATA_TABLE_PATTERN.matcher(serializedTableId);
+    String project = matcher.group("project");
+    String dataset = matcher.group("dataset");
+    String table = matcher.group("table");
+
+    if (project == null || dataset == null || table == null) {
+      logger.error("Found blob `{}/{}` without parsable table metadata.",
+                   blob.getBucket(), blob.getName());
+      return null;
+    }
+
+    return TableId.of(project, dataset, table);
+  }
+
+  private Map<Job, List<Blob>> triggerBigQueryLoadJobs(Map<TableId, List<Blob>> tablesToBlobs) {
+    Map<Job, List<Blob>> newJobs = new HashMap<>(tablesToBlobs.size());
+    for (Map.Entry<TableId, List<Blob>> entry : tablesToBlobs.entrySet()) {
+      newJobs.put(triggerBigQueryLoadJob(entry.getKey(), entry.getValue()), entry.getValue());
+    }
+    return newJobs;
+  }
+
+  private Job triggerBigQueryLoadJob(TableId table, List<Blob> blobs) {
+    List<String> uris = blobsToURIs(blobs);
+    // create job load configuration
+    LoadJobConfiguration loadJobConfiguration =
+        LoadJobConfiguration.newBuilder(table, uris)
+            .setFormatOptions(FormatOptions.json())
+            .setCreateDisposition(JobInfo.CreateDisposition.CREATE_IF_NEEDED)
```

Comment:
According to https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs the default write disposition is WRITE_IF_EMPTY. I think we might need to set write disposition to WRITE_APPEND. Is it possible to do this in the builder?

Reply:
yep. (and that seems like an oversight, so ty for pointing it out)